src/components/AdminSettings/AllowedGroups.vue

```diff
@@ -131,11 +135,15 @@ export default {
 
 	mounted() {
 		this.loading = true
-		this.allowedGroups = loadState('talk', 'allowed_groups')
-		this.canStartConversations = loadState('talk', 'start_conversations')
+		this.allowedGroups = loadState('talk', 'allowed_groups').sort(function(a, b) {
+			return a.displayname.localeCompare(b.displayname)
+		})
+		this.canStartConversations = loadState('talk', 'start_conversations').sort(function(a, b) {
+			return a.displayname.localeCompare(b.displayname)
+		})
 		this.startCalls = startCallOptions[parseInt(loadState('talk', 'start_calls'))]
-		this.groups = [...new Set(this.allowedGroups.concat(this.canStartConversations))].sort(function(a, b) {
-			return a.localeCompare(b)
+		this.groups = Array.from(new Set(this.allowedGroups.concat(this.canStartConversations).map(g => JSON.stringify(g)))).map(g => JSON.parse(g)).sort(function(a, b) {
```

Comment:
some weird gymnastics here, I'm not really sure why

either move this into a function with an explicit name or add a code comment, if there is no better way

Reply:
I'm bad at JS and just copied SO until it worked.
we have two arrays with objects (id + name) and want to merge them into a list without duplicates.

Reply:
the Set thing is fine.
I was mostly wondering about the converting to JSON string then parsing again.

ok, let's try and make it more readable to understand it:
```
Array.from(
   new Set(
      this.allowedGroups.concat(this.canStartConversations)
         .map(g => JSON.stringify(g))
   )
)
.map(g => JSON.parse(g))
.sort(...)
```

hmmmm, so we are trying to make the list unique by using the JSON **as string** as the key for uniqueness, so it only works with the assumption that the JSON object's key order is always the same

I guess the alternative would be to construct an object where keys are the id and values the JSON object, for deduplicating by id, and then retrieving only the values.

Either way I guess the approach is fine, please at least add a comment to clarify deduplication to avoid future head scratching.